ipsource.go

```diff
@@ -307,6 +308,74 @@ func (u NetInterface) GetIPs(ctx context.Context, versions IPVersions) ([]net.IP
 	return ips, nil
 }
 
+type Static struct {
+	IPs []net.IP `json:"ips,omitempty"`
+
+	logger      *zap.Logger
+	invalid_ips []string
+}
+
+func (Static) CaddyModule() caddy.ModuleInfo {
+	return caddy.ModuleInfo{
+		ID:  "dynamic_dns.ip_sources.static",
+		New: func() caddy.Module { return new(Static) },
+	}
+}
+
+func (s Static) GetIPs(ctx context.Context, versions IPVersions) ([]net.IP, error) {
+	if versions.V4Enabled() && versions.V6Enabled() {
+		return s.IPs, nil
+	}
+
+	ips := []net.IP{}
+
+	for _, ip := range s.IPs {
+		if versions.V4Enabled() && ip.To4() != nil {
+			ips = append(ips, ip)
+			continue
+		}
+
+		if versions.V6Enabled() && ip.To16() != nil {
+			ips = append(ips, ip)
+			continue
+		}
+	}
+
+	return ips, nil
+}
+
+func (s *Static) Provision(ctx caddy.Context) error {
+	s.logger = ctx.Logger(s)
+
+	for _, invalid_ip := range s.invalid_ips {
+		s.logger.Warn("Invalid IP", zap.String("invalid_ip", invalid_ip))
+	}
+
+	if s.IPs == nil {
+		s.logger.Warn("No static IPs configured")
+	}
+
+	return nil
+}
+
+func (s *Static) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
+	d.Next() // skip directive name
+
+	for d.NextArg() {
+		raw_ip := d.Val()
+		ip := net.ParseIP(raw_ip)
+
+		if ip == nil {
+			s.invalid_ips = append(s.invalid_ips, raw_ip)
+			continue
+		}
```

Comment:
This is weird IMO, just return an error if any invalid IPs are configured.

Reply:
Is this better?

Reply:
I think so!